main.go

```diff
@@ -617,13 +621,25 @@ func (c *config) rewrite(node ast.Node, start, end int) (ast.Node, error) {
 				continue
 			}
 
-			if f.Tag == nil {
-				f.Tag = &ast.BasicLit{}
-			}
-
 			fieldName := ""
 			if len(f.Names) != 0 {
-				fieldName = f.Names[0].Name
+				if c.skipPrivateFields {
+				publicFieldSearch:
+					for _, field := range f.Names {
+						for _, c := range field.Name {
+							if unicode.IsUpper(c) {
+								fieldName = field.Name
+								break publicFieldSearch
+							}
+							break
+						}
+					}
+					if fieldName == "" {
+						continue
+					}
+				} else {
+					fieldName = f.Names[0].Name
+				}
```

Comment:
I don't like using labels, but I think we can still achieve it by using an anonymous function. It also reads easier:

```diff
diff --git a/main.go b/main.go
index c7e0276..14b8a7b 100644
--- a/main.go
+++ b/main.go
@@ -608,6 +608,19 @@ func (c *config) offsetSelection(file ast.Node) (int, int, error) {
 func (c *config) rewrite(node ast.Node, start, end int) (ast.Node, error) {
        errs := &rewriteErrors{errs: make([]error, 0)}

+       processField := func(name string) bool {
+               if !c.skipPrivateFields {
+                       return true
+               }
+
+               for _, r := range name {
+                       if unicode.IsUpper(r) {
+                               return true
+                       }
+               }
+               return false
+       }
+
        rewriteFunc := func(n ast.Node) bool {
                x, ok := n.(*ast.StructType)
                if !ok {
@@ -623,22 +636,15 @@ func (c *config) rewrite(node ast.Node, start, end int) (ast.Node, error) {

                        fieldName := ""
                        if len(f.Names) != 0 {
-                               if c.skipPrivateFields {
-                               publicFieldSearch:
-                                       for _, field := range f.Names {
-                                               for _, c := range field.Name {
-                                                       if unicode.IsUpper(c) {
-                                                               fieldName = field.Name
-                                                               break publicFieldSearch
-                                                       }
-                                                       break
-                                               }
-                                       }
-                                       if fieldName == "" {
-                                               continue
+                               for _, field := range f.Names {
+                                       if processField(field.Name) {
+                                               fieldName = field.Name
                                        }
-                               } else {
-                                       fieldName = f.Names[0].Name
+                               }
+
+                               // nothing to process, continue with next line
+                               if fieldName == "" {
+                                       return true
                                }
                        }
```

Reply:
It seems more logical to me to extract `isPublicName` function.
Could you please review the version that I've just pushed.
Your additional comment "nothing to process, continue with next line" is incorporated.